SIPS/sip-12.md

```diff
@@ -51,12 +51,18 @@ This permission is specified as follows in `snap.manifest.json` files:
   "initialPermissions": {
     "endowment:name-lookup": {
         "chains": ["eip155:1", "bip122:000000000019d6689c085ae165831e93"],
+        "matchers": { "tlds": ["lens"], "schemes": ["farcaster"] }
     }
   }
 }
 ```
 
-`chains` - An array of CAIP-2 chain IDs that the snap supports. This field is useful for a client in order to avoid unnecessary overhead.
+`chains` - A non-empty array of CAIP-2 chain IDs that the snap supports. This field is useful for a client in order to avoid unnecessary overhead.
+`matchers` - A non-empty object that MUST contain 1 or both of the below properties. These matchers are useful for a client for validating input for domain resolution, also helpful in reducing overhead. 
```

Comment:
@hmalik88 Wouldn't it be better for `matchers` to accept regex of what the SNAP can resolve rather than tlds/schemes, which may be limiting.
Example:
* TLDs: ^[a-z0-9]+\.lens$
* schemes: ^farcaster:[a-z0-9]+$

This would allow for far more flexibility (not limit naming systems to predetermined formats) and accuracy (some naming systems only allow ASCII) and not require the user to know what is the prefix of the naming system they are using.

Reply:
We decided against accepting regex because of [ReDoS](https://owasp.org/www-community/attacks/Regular_expression_Denial_of_Service_-_ReDoS)

Reply:
Made `matchers` optional 🤝

Reply:
@hmalik88 Thank you! For planning purposes, any idea when this will be available in production Metamask, or where to ask?

Reply:
Soon-ish, I'll be working to finish this in the current or next sprint.